quickwit/quickwit-query/src/query_ast/range_query.rs

```diff
@@ -191,13 +220,33 @@ impl BuildTantivyAst for RangeQuery {
                 if let Some(range) = bounds_range {
                     return Ok(query_from_fast_val_range(&empty_term, range).into());
                 }
-                // TODO add support for str query
-                return Ok(EmptyQuery.into());
+
+                let mut normalizer = options
+                    .get_fast_field_tokenizer_name()
+                    .and_then(|tokenizer_name| tokenizer_manager.get_normalizer(tokenizer_name));
+
+                let (lower_bound, upper_bound) =
+                    convert_bounds(&self.lower_bound, &self.upper_bound, field_entry.name())?;
+                FastFieldRangeQuery::new(
+                    lower_bound.map(|val| {
+                        let val = get_normalized_text(&mut normalizer, val);
+                        let mut term = empty_term.clone();
+                        term.append_type_and_str(&val);
+                        term
+                    }),
+                    upper_bound.map(|val| {
+                        let val = get_normalized_text(&mut normalizer, val);
+                        let mut term = empty_term.clone();
+                        term.append_type_and_str(&val);
+                        term
+                    }),
+                )
+                .into()
```

Comment:
this has a funky behavior in case the bounds are numbers, but the column contains at least some strings. For instance if have `{"abc": "1"}`, `abc:1` returns it, but `abc:[0 TO 2]` returns nothing, while `abc:[0 TO :]` does actually returns something
I think for json fields, it would be safer to build an OR of integer/string if the bounds can be parsed to an integer